eZ/Publish/API/Repository/Tests/ContentTypeServiceTest.php

```diff
@@ -1416,6 +1416,43 @@ public function testUpdateContentTypeDraftWithNewTranslation()
         );
     }
 
+    /**
+     * @covers \eZ\Publish\API\Repository\ContentTypeService::updateContentTypeDraft
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testUpdateContentTypeWithDescriptionClearing()
+    {
+        $repository = $this->getRepository();
+        $contentTypeService = $repository->getContentTypeService();
+
+        $contentTypeDraft = $this->createContentTypeDraft();
+        $contentTypeService->publishContentTypeDraft($contentTypeDraft);
+
+        $contentType = $contentTypeService->loadContentType($contentTypeDraft->id);
+        // sanity check
+        self::assertEquals(
+            ['eng-US', 'ger-DE'],
+            array_keys($contentType->getNames())
+        );
+
+        $contentTypeDraft = $contentTypeService->createContentTypeDraft($contentType);
+        $updateStruct = $contentTypeService->newContentTypeUpdateStruct();
+        $updateStruct->languageCode = 'ger-DE';
+        $updateStruct->descriptions = null;
+        $contentTypeService->updateContentTypeDraft($contentTypeDraft, $updateStruct);
+        $contentTypeService->publishContentTypeDraft($contentTypeDraft);
```

Comment:
The Update Struct behaves now in a weird way. It accepts all the `names` and all the `descriptions`, but allows setting `languageCode`. It's a bit confusing. Since the `languageCode` exists now, my expectation would be that when I set it, only the names and descriptions of that language are affected. Yet, it doesn't work like that (I'm still able to update non-German description after setting `ger-DE` as language Code). Moreover behavior for names hasn't changed, so it's also the same case.

The design of AdminUI is a bit faulty here as it assumes only one translation is updated at a time. It's probably also my mistake not to pay closer attention to that design when it was created.

The possible directions I see here are:
1. Make the behavior uniform - affect only the given translation both for removal and update and both for names and description (of course we cannot clear name - it's required AFAIR).
2. Make language code tied to a new set of properties only - `name` and `description` so it's clear what to update and possibly deprecate names.
3. Instead of trying to fix here faulty logic of AdminUI, pass from AdminUI what the Repository expects - empty value for a translation which was unset. For a database space optimization I can accept traversing through both descriptions to unset those which are `null` - as an indication they should not exist. Empty strings should be serialized as-is.

The test should cover more use cases:
- keep all descriptions unchanged (see comment for `...->descriptions = null` line)
- clear description of a single translation
- clear descriptions of all translations
- update description of a single translation
- update descriptions of all translations

A data provider would be good in this case.

Reply:
@alongosz I think it could be done even easier: https://github.com/ezsystems/repository-forms/pull/338 - for eZ Platform 2.5 - (`admin-ui` PR should be created accordingly for 3.0).

Probably this PR could be closed, as it really is `repository-forms`/`admin-ui` problem and it also handles clearing description fields for `ContentType's` fieldtypes.